frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/GitFetchSupportImpl.java

```diff
@@ -73,14 +73,15 @@ private GitFetchSupportImpl(Project project) {
     this.project = project;
   }
 
-  private static final AtomicReference<@Nullable GitFetchSupportImpl> INSTANCE = new AtomicReference<>(null);
+  private static final AtomicReference<@Nullable GitFetchSupportImpl> instance = new AtomicReference<>(null);
+  private static final Object LOCK = new Object();
 
-  public static GitFetchSupportImpl fetchSupport(Project project) {
-    var gitFetchSupport = INSTANCE.get();
+  @SuppressWarnings("regexp")
+  public static synchronized GitFetchSupportImpl fetchSupport(Project project) {
+    var gitFetchSupport = instance.get();
```

Comment:
I don't think `AtomicReference` is needed now that the access to the field is only via a statically synchronized method